### PR TITLE
Shows a better error message instead of a NPE

### PR DESCRIPTION
**Pull request by shubham1g5**
Trying to display a better error on getting a 500 from Server during app update. Earlier app was failing with this error message - 
"Attempt to invoke virtual method 'java.io/InputStream okhttp3.ResponseBody.byteStream()' on a null object reference".

### CODE REVIEW

File: app/src/org/commcare/engine/references/JavaHttpReference.java [hunk 1]
```diff
@@ -35,7 +39,12 @@ public OutputStream getOutputStream() throws IOException {
 
     @Override
     public InputStream getStream() throws IOException {
-        return generator.simpleGet(uri).body().byteStream();
+        Response<ResponseBody> response = generator.simpleGet(uri);
+        if (response.isSuccessful()) {
+            return response.body().byteStream();
+        } else {
+            throw new RuntimeException(Localization.get("install.fail.error", Integer.toString(response.code())));
```
Comment (wpride) [inline app/src/org/commcare/engine/references/JavaHttpReference.java]: Granted that we expect HTTP errors to happen fairly regularly is throwing a `RuntimeException` the best behavior? Can we throw this as an `IOException` and catch that ? 

Also, won't this crash the application?

Comment (shubham1g5) [reply app/src/org/commcare/engine/references/JavaHttpReference.java]: We are catching the generic Exception [here](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/tasks/UpdateTask.java#L125) so it doesn't crash the app though can definitely wrap it as `IOException` if it's more clean.

Comment (wpride) [reply app/src/org/commcare/engine/references/JavaHttpReference.java]: Ah gotcha, nice. Yeah I think adding a block for `IOException` there handling this case would be preferable, but could be just my taste

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Pull request merged**